### PR TITLE
on MacOS, Fix not sending ReceivedCharacter event for some key combination

### LINKED ISSUE #1267: MacOS not sending ReceivedCharacter event
It has been reported to Alacritty that on macOS, specific key combinations do not always emit `ReceivedCharacter` events on macOS.

The specific example was Ctrl+Q and it appears that while the first press does not emit an event, the second does.

This has been reported to Alacritty in https://github.com/jwilm/alacritty/issues/1359 and here's an excerpt from the relevant event log (full log can be found in the issue):

```
[2019-11-08 10:53] [INFO] glutin event: WindowEvent { window_id: WindowId(Id(140382647460256)), event: KeyboardInput { device_id: DeviceId(DeviceId), input: KeyboardInput { scancode: 62, state: Pressed, virtual_keycode: Some(RControl), modifiers: ModifiersState { shift: false, ctrl: true, alt: false, logo: false } } } }
[2019-11-08 10:53] [INFO] glutin event: WindowEvent { window_id: WindowId(Id(140382647460256)), event: KeyboardInput { device_id: DeviceId(DeviceId), input: KeyboardInput { scancode: 12, state: Pressed, virtual_keycode: Some(Q), modifiers: ModifiersState { shift: false, ctrl: true, alt: false, logo: false } } } }
[2019-11-08 10:53] [INFO] glutin event: WindowEvent { window_id: WindowId(Id(140382647460256)), event: KeyboardInput { device_id: DeviceId(DeviceId), input: KeyboardInput { scancode: 12, state: Released, virtual_keycode: Some(Q), modifiers: ModifiersState { shift: false, ctrl: true, alt: false, logo: false } } } }
[2019-11-08 10:53] [INFO] glutin event: WindowEvent { window_id: WindowId(Id(140382647460256)), event: KeyboardInput { device_id: DeviceId(DeviceId), input: KeyboardInput { scancode: 62, state: Released, virtual_keycode: Some(RControl), modifiers: ModifiersState { shift: false, ctrl: false, alt: false, logo: false } } } }
[2019-11-08 10:53] [INFO] glutin event: WindowEvent { window_id: WindowId(Id(140382647460256)), event: KeyboardInput { device_id: DeviceId(DeviceId), input: KeyboardInput { scancode: 62, state: Pressed, virtual_keycode: Some(RControl), modifiers: ModifiersState { shift: false, ctrl: true, alt: false, logo: false } } } }
[2019-11-08 10:53] [INFO] glutin event: WindowEvent { window_id: WindowId(Id(140382647460256)), event: KeyboardInput { device_id: DeviceId(DeviceId), input: KeyboardInput { scancode: 12, state: Pressed, virtual_keycode: Some(Q), modifiers: ModifiersState { shift: false, ctrl: true, alt: false, logo: false } } } }
[2019-11-08 10:53] [INFO] glutin event: WindowEvent { window_id: WindowId(Id(140382647460256)), event: ReceivedCharacter('\u{11}') }
[2019-11-08 10:53] [INFO] glutin event: WindowEvent { window_id: WindowId(Id(140382647460256)), event: KeyboardInput { device_id: DeviceId(DeviceId), input: KeyboardInput { scancode: 12, state: Released, virtual_keycode: Some(Q), modifiers: ModifiersState { shift: false, ctrl: true, alt: false, logo: false } } } }
[2019-11-08 10:53] [INFO] glutin event: WindowEvent { window_id: WindowId(Id(140382647460256)), event: KeyboardInput { device_id: DeviceId(DeviceId), input: KeyboardInput { scancode: 62, state: Released, virtual_keycode: Some(RControl), modifiers: ModifiersState { shift: false, ctrl: false, alt: false, logo: false } } } }
```

### PR DESCRIPTION
**Pull request by hatoo**
- [x] Tested on all platforms changed
- [x] Compilation warnings were addressed
- [x] `cargo fmt` has been run on this branch
- [x] `cargo doc` builds successfully
- [x] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [ ] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [ ] Created or updated an example program if it would help users understand this functionality
- [ ] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

This PR fixes #1267 
It seems that Ctrl+Q is some prefix key when `hasMarkedText` returns `NO`.
For example, it emits '\u{1e}' when input Ctrl+Q and Arrow-up key.
So I modified `hasMarkedText` to returns YES always.
I'm not 100% sure about the mean of this change, but I think it's OK because `hasMarkedText` returns always `YES` after inputting Option+N in previous implementation.

A key with Command no longer emits ReceivedCharacter event by this change.

### CODE REVIEW

File: src/platform_impl/macos/view.rs [hunk 1]
```diff
@@ -357,13 +357,10 @@ extern "C" fn reset_cursor_rects(this: &Object, _sel: Sel) {
...
-extern "C" fn has_marked_text(this: &Object, _sel: Sel) -> BOOL {
-    unsafe {
-        trace!("Triggered `hasMarkedText`");
-        let marked_text: id = *this.get_ivar("markedText");
-        trace!("Completed `hasMarkedText`");
-        (marked_text.length() > 0) as i8
-    }
+extern "C" fn has_marked_text(_this: &Object, _sel: Sel) -> BOOL {
+    trace!("Triggered `hasMarkedText`");
+    trace!("Completed `hasMarkedText`");
```
Comment (Osspial) [inline src/platform_impl/macos/view.rs]: `hasMarkedText` is no longer getting triggered, so these logging calls shouldn't be necessary

**Comment by chrisduerr** [comment]
I was not able to reproduce, but @jeremydr2 was able to share an event log with the missing events. Maybe he can create another log by running `cargo run --example window` in this repo for comparison to see if it got fixed?

**Comment by jeremydr2** [comment]
I hit C-q once. Here are the logs:

```
NewEvents(WaitCancelled { start: Instant { t: 202776246335713 }, requested_resume: None })
2020-01-14 13:32:48,664 TRACE [winit::platform_impl::platform::view] Triggered `flagsChanged`
2020-01-14 13:32:48,664 TRACE [winit::platform_impl::platform::view] Completed `flagsChanged`
WindowEvent { window_id: WindowId(Id(140631488875984)), event: KeyboardInput { device_id: DeviceId(DeviceId), input: KeyboardInput { scancode: 62, state: Pressed, virtual_keycode: Some(RControl), modifiers: CTRL }, is_synthetic: false } }
DeviceEvent { device_id: DeviceId(DeviceId), event: ModifiersChanged(CTRL) }
MainEventsCleared
RedrawRequested(WindowId(Id(140631488875984)))
RedrawEventsCleared
NewEvents(WaitCancelled { start: Instant { t: 202778853666746 }, requested_resume: None })
2020-01-14 13:32:48,706 TRACE [winit::platform_impl::platform::view] Triggered `keyDown`
2020-01-14 13:32:48,707 TRACE [winit::platform_impl::platform::view] Triggered `hasMarkedText`
2020-01-14 13:32:48,707 TRACE [winit::platform_impl::platform::view] Completed `hasMarkedText`
2020-01-14 13:32:48,707 TRACE [winit::platform_impl::platform::view] Triggered `hasMarkedText`
2020-01-14 13:32:48,707 TRACE [winit::platform_impl::platform::view] Completed `hasMarkedText`
2020-01-14 13:32:48,707 TRACE [winit::platform_impl::platform::view] Completed `keyDown`
WindowEvent { window_id: WindowId(Id(140631488875984)), event: KeyboardInput { device_id: DeviceId(DeviceId), input: KeyboardInput { scancode: 12, state: Pressed, virtual_keycode: Some(Q), modifiers: CTRL }, is_synthetic: false } }
MainEventsCleared
RedrawRequested(WindowId(Id(140631488875984)))
RedrawEventsCleared
NewEvents(WaitCancelled { start: Instant { t: 202778896226323 }, requested_resume: None })
2020-01-14 13:32:48,832 TRACE [winit::platform_impl::platform::view] Triggered `keyUp`
2020-01-14 13:32:48,832 TRACE [winit::platform_impl::platform::view] Completed `keyUp`
WindowEvent { window_id: WindowId(Id(140631488875984)), event: KeyboardInput { device_id: DeviceId(DeviceId), input: KeyboardInput { scancode: 12, state: Released, virtual_keycode: Some(Q), modifiers: CTRL }, is_synthetic: false } }
MainEventsCleared
RedrawRequested(WindowId(Id(140631488875984)))
RedrawEventsCleared
NewEvents(WaitCancelled { start: Instant { t: 202779021806898 }, requested_resume: None })
2020-01-14 13:32:48,855 TRACE [winit::platform_impl::platform::view] Triggered `flagsChanged`
2020-01-14 13:32:48,855 TRACE [winit::platform_impl::platform::view] Completed `flagsChanged`
WindowEvent { window_id: WindowId(Id(140631488875984)), event: KeyboardInput { device_id: DeviceId(DeviceId), input: KeyboardInput { scancode: 62, state: Released, virtual_keycode: Some(RControl), modifiers: (empty) }, is_synthetic: false } }
DeviceEvent { device_id: DeviceId(DeviceId), event: ModifiersChanged((empty)) }
MainEventsCleared
RedrawRequested(WindowId(Id(140631488875984)))
```

**Comment by chrisduerr** [comment]
Thanks a lot for testing, looks like this should be good to go then!

**Pull request merged**